kafka/pykafka/consumer.py

```diff
@@ -29,15 +133,63 @@ def topic(self):
     def partitions(self):
         return self._partitions
 
+    @property
+    def fetch_message_max_bytes(self):
+        return self._fetch_message_max_bytes
+
     def __iter__(self):
         while True:
-            yield self.consume()
+            message = self.consume(timeout=self._socket_timeout_ms)
+            if not message and self._consumer_timed_out():
+                raise StopIteration
+            yield message
+
+    def _consumer_timed_out(self):
+        if self._consumer_timeout_ms == -1:
+            return False
+        disp = (time.time() - self._last_message_time) * 1000.0
+        return disp > self._consumer_timeout_ms
 
     def consume(self, timeout=None):
         """Get one message from the consumer.
 
         :param timeout: Seconds to wait before returning None
         """
+        owned_partition = self.partition_cycle.next()
+        message = owned_partition.consume(timeout=timeout)
+
+        if self._auto_commit_enable:
+            self._auto_commit()
```

Comment:
The consumer is going to need to get a `cluster.Cluster` instance in the constructor and then you can do `self.cluster.handler.spawn`.  The original intent of those handlers, carried forward from samsa, is that it'll let you switch between `threading` and `gevent` seamlessly.